pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -544,16 +544,30 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 			return nil, fmt.Errorf("pod %q container %q changed request for resource %q from %d to %d", string(podUID), contName, resource, devices.Len(), required)
 		}
 	}
-	if needed == 0 {
-		// No change, no work.
-		return nil, nil
-	}
+
 	klog.V(3).InfoS("Need devices to allocate for pod", "deviceNumber", needed, "resourceName", resource, "podUID", string(podUID), "containerName", contName)
+	healthyDevices, hasRegistered := m.healthyDevices[resource]
+
 	// Check if resource registered with devicemanager
-	if _, ok := m.healthyDevices[resource]; !ok {
+	if !hasRegistered {
 		return nil, fmt.Errorf("can't allocate unregistered device %s", resource)
 	}
 
+	// Check if registered resource has healthy devices
+	if healthyDevices.Len() == 0 {
+		return nil, fmt.Errorf("can't allocate unhealthy devices %s", resource)
```

Comment:
This error message doesn't look correct for a zero length check.

Reply:
This check is to ensure that we rule out cases where we don't have healthy devices that are suitable for allocation, hence the message. Would you prefer it to be more descriptive such as  `fmt.Errorf("no healthy devices present, can't allocate unhealthy devices %s", resource)`?

Reply:
I think that is very helpful!

Reply:
Updated with a more descriptive error message.